client2/arq.go

```diff
@@ -0,0 +1,241 @@
+// SPDX-FileCopyrightText: © 2023 David Stainton
+// SPDX-License-Identifier: AGPL-3.0-only
+package client2
+
+import (
+	"errors"
+	"io"
+	"sync"
+	"time"
+
+	"github.com/charmbracelet/log"
+
+	"github.com/katzenpost/katzenpost/core/crypto/rand"
+	sConstants "github.com/katzenpost/katzenpost/core/sphinx/constants"
+)
+
+const (
+	// MessageIDLength is the length of a message ID in bytes.
+	MessageIDLength = 16
```

Comment:
Who uses this message ID, and who generates it? Is it unique to the ARQ stuff or to the client2  stack or is it a deeper concept?